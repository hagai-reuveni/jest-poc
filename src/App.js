import React, { Component } from 'react';
import Routes from './Routes';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import './App.css';

class App extends Component {
  render() {
    return (
    <Router>
      <main className="container">
        <div>
          <h1>This is a jest poc test</h1>
          <p>If you see this everything is working!</p>
        </div>
        <Routes />
      </main>
    </Router>
    );
  }
}

export default App;
