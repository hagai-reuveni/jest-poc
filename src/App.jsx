import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import Routes from './Routes';
import { NavLink } from 'react-router-dom'

import './App.css';

export default () => (
  <Router>
    <div>
      <nav>
        <NavLink to="/">Home</NavLink> | | <NavLink to="/users">Users</NavLink>
      </nav>
      <main className="container">
        <div>
          <h1>This is a jest poc test</h1>
          <p>If you see this everything is working!</p>
        </div>
        <Routes />
      </main>
    </div>
  </Router>
  );
