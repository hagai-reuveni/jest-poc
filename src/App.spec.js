import React from 'react';
import App from './App';

describe('<App />', () => {
  it('renders without crashing', () => {
    const container = shallow(<App />);
    expect(container).toMatchSnapshot();
  });

  it('should render an `h1', () => {
    const container = shallow(<App />);
    expect(container.find('h1').length).toBe(1);
    expect(container.find('h1').text()).toEqual('This is a jest poc test');
  });
});
//console.log(window); // this is to test if jest is running with js-dom