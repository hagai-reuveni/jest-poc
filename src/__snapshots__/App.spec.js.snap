// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
<BrowserRouter>
  <div>
    <nav>
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/"
      >
        Home
      </NavLink>
       | | 
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/users"
      >
        Users
      </NavLink>
    </nav>
    <main
      className="container"
    >
      <div>
        <h1>
          This is a jest poc test
        </h1>
        <p>
          If you see this everything is working!
        </p>
      </div>
      <Routes />
    </main>
  </div>
</BrowserRouter>
`;
